bindings/python/pytpm2tss/__init__.py

```diff
@@ -0,0 +1,2533 @@
+                               authHandle.handle,
+                               session1, session2, session3))
+        return()
+
+    def ClearControl(self,auth, disable,
+                     session1=None, session2=None, session3=None):
+        session1 = session1.handle if session1 else _lib.ESYS_TR_NONE
+        session2 = session2.handle if session2 else _lib.ESYS_TR_NONE
+        session3 = session3.handle if session3 else _lib.ESYS_TR_NONE
+        _chkrc(_lib.Esys_ClearControl(self.ctx,
+                                      auth.handle,
+                                      session1, session2, session3,
+                                      disable))
+        return()
+
+    def HierarchyChangeAuth(self,authHandle, newAuth,
+                            session1=None, session2=None, session3=None):
+        session1 = session1.handle if session1 else _lib.ESYS_TR_NONE
+        session2 = session2.handle if session2 else _lib.ESYS_TR_NONE
+        session3 = session3.handle if session3 else _lib.ESYS_TR_NONE
+        _chkrc(_lib.Esys_HierarchyChangeAuth(self.ctx,
+                                             authHandle.handle,
+                                             session1, session2, session3,
+                                             newAuth))
+        return()
+
+    def DictionaryAttackLockReset(self,lockHandle,
+                                  session1=None, session2=None, session3=None):
+        session1 = session1.handle if session1 else _lib.ESYS_TR_NONE
+        session2 = session2.handle if session2 else _lib.ESYS_TR_NONE
+        session3 = session3.handle if session3 else _lib.ESYS_TR_NONE
+        _chkrc(_lib.Esys_DictionaryAttackLockReset(self.ctx,
+                                                   lockHandle.handle,
+                                                   session1, session2, session3))
+        return()
+
+    def DictionaryAttackParameters(self,lockHandle, newMaxTries, newRecoveryTime, lockoutRecovery,
+                                   session1=None, session2=None, session3=None):
+        session1 = session1.handle if session1 else _lib.ESYS_TR_NONE
+        session2 = session2.handle if session2 else _lib.ESYS_TR_NONE
+        session3 = session3.handle if session3 else _lib.ESYS_TR_NONE
+        _chkrc(_lib.Esys_DictionaryAttackParameters(self.ctx,
+                                                    lockHandle.handle,
+                                                    session1, session2, session3,
+                                                    newMaxTries,
+                                                    newRecoveryTime,
+                                                    lockoutRecovery))
+        return()
+
+    def PP_Commands(self,auth, setList, clearList,
+                    session1=None, session2=None, session3=None):
+        session1 = session1.handle if session1 else _lib.ESYS_TR_NONE
+        session2 = session2.handle if session2 else _lib.ESYS_TR_NON
```

Comment:
I don't know what you are referring to with "this". Could you elaborate ?

Reply:
ok, I got it now, damn code-viel in here... :-(

Reply:
Same problem as below. This comment was probably meant for `TPM2_ALG`, but also applies to similar classes such as `TPM2_CLOCK`. Using the Enum classes allows for some type safety to be added, since you cannot really use a `TPM2_CLOCK` value in place of a `TPM2_ALG` value (even though the underlying C types might be compatible).